audbackend/core/artifactory.py

```diff
@@ -259,6 +267,30 @@ def _get_file(
         src_path = self._path(src_path, version)
         _download(src_path, dst_path, verbose=verbose)
 
+    def _legacy_split_ext(
+            self,
+            name: str,
+    ) -> typing.Tuple[str, str]:
+        r"""Split name into basename and extension."""
+
+        ext = None
+        for custom_ext in self._legacy_extensions:
```

Comment:
Above you set the default value to
```python
self._legacy_extensions = None
```
so it seems a little bit risky to just loop it here. We should switch to `[]` as default or add a check here if `self._legacy_extensions` is `None`

Reply:
Now I found that `_use_legacy_file_structure()` will ensure that it is never set to `None` when the legacy paths are used. But maybe it will still be easier to understand of the default is `[]` or to we gain some additional information by using `None` instead?

Reply:
I changed to `[]` as default value.